### PR TITLE
[Part 3][IT tests] JAMES-2717 Replace the remaining embedded ES components in IT tests

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[5b5008cac5842adead8c19e2d4fd08662e539f41] [ERROR] Errors: 
[5b5008cac5842adead8c19e2d4fd08662e539f41] [ERROR]   CassandraSetMessagesMethodCucumberTest » NullPointer

[5b5008cac5842adead8c19e2d4fd08662e539f41] [ERROR] A user can destroy a draft  Time elapsed: 2.254 s  <<< ERROR!
[5b5008cac5842adead8c19e2d4fd08662e539f41] java.lang.NullPointerException
```

**Comment by trantienduchn** [comment]
:green_apple:

**Comment by trantienduchn** [comment]
```
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Apache James :: Server :: Mailets Integration Testing SUCCESS [02:10 min]
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Apache James :: Server :: JMAP :: Cassandra Integration testing SUCCESS [05:44 min]
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Apache James :: Server :: JMAP :: Memory Integration testing SUCCESS [04:57 min]
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Apache James :: Server :: JMAP :: RabbitMQ + Object Store + Cassandra Integration testing SUCCESS [07:51 min]
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Apache James :: Server :: Web Admin server integration tests SUCCESS [04:37 min]
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Apache James :: Server :: Web Admin 3.4.0-SNAPSHOT . SUCCESS [  0.627 s]
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] ------------------------------------------------------------------------
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] BUILD SUCCESS
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] ------------------------------------------------------------------------
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Total time: 02:23 h
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] Finished at: 2019-05-09T06:45:40Z
[c3c42b5dac803a0f90367ccaa2920b9b126baf50] [INFO] ------------------------------------------------------------------------
```

File: server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/cucumber/CucumberElasticSearchSingleton.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.jmap.cassandra.cucumber;
+
+import org.apache.james.DockerElasticSearchRule;
+
+public class CucumberElasticSearchSingleton {
+
+    public static DockerElasticSearchRule elasticSearch = new DockerElasticSearchRule();
```
Comment (remk) [inline server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/cucumber/CucumberElasticSearchSingleton.java]: final ?

File: server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/cucumber/CassandraStepdefs.java [hunk 2]
```diff
@@ -56,7 +54,7 @@
     private final MainStepdefs mainStepdefs;
     private final ImapStepdefs imapStepdefs;
     private TemporaryFolder temporaryFolder = new TemporaryFolder();
-    private EmbeddedElasticSearch embeddedElasticSearch = new EmbeddedElasticSearch(temporaryFolder);
+    private DockerElasticSearchRule elasticSearch = CucumberElasticSearchSingleton.elasticSearch;
```
Comment (remk) [inline server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/cucumber/CassandraStepdefs.java]: final ?

File: server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/cucumber/awss3/CucumberElasticSingleton.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.jmap.rabbitmq.cucumber.awss3;
+
+import org.apache.james.DockerElasticSearchRule;
+
+class CucumberElasticSingleton {
+
+    static DockerElasticSearchRule elasticSearch = new DockerElasticSearchRule();
```
Comment (remk) [inline server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/java/org/apache/james/jmap/rabbitmq/cucumber/awss3/CucumberElasticSingleton.java]: final ?

**Comment by trantienduchn** [comment]
was :green_apple: just rebased

**Comment by trantienduchn** [comment]
rebase because of the compling issue

**Comment by trantienduchn** [comment]
```
[fa12101dd7cda252d35498d94118fa6da5a3510d] [ERROR]   ManageableMailQueueContract.dequeueShouldDecreaseQueueSize:82 » AlreadyClosed ...
```
Not related, test this please

**Comment by trantienduchn** [comment]
```
[fa12101dd7cda252d35498d94118fa6da5a3510d] [ERROR]   ManageableMailQueueContract.browseShouldReturnMailsWithAttributes:475 » NoSuchElement
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**